eacl/validator.go

```diff
@@ -152,8 +152,15 @@ func targetMatches(unit *ValidationUnit, record *Record) bool {
 		// check public key match
 		if pubs := target.BinaryKeys(); len(pubs) != 0 {
 			for _, key := range pubs {
-				if bytes.Equal(key, unit.key) {
-					return true
+				switch len(key) {
+				case 25: // user.ID.
+					if len(unit.account) > 0 && bytes.Equal(key, unit.account) {
```

Comment:
Why `len()` check here?

Reply:
@cthulhu-rider [worried](https://github.com/nspcc-dev/neofs-sdk-go/pull/592#discussion_r1666737509) about the case when unit.account is unset and key is empty

Reply:
> and key is empty

but now u use switch-case so key is definitely not empty

Reply:
Removed